In [4]:
import pandas as pd
import matplotlib
dados = pd.read_csv('TabelaSalarial.csv')
dados.head()
dados.describe()


N  N de Filhos  Salario (x Sal Min)  Idade (em anos)
count  36.000000    20.000000            36.000000        36.000000
mean   18.500000     1.650000            11.122222        34.583333
std    10.535654     1.268028             4.587458         6.737422
min     1.000000     0.000000             4.000000        20.000000
25%     9.750000     1.000000             7.552500        30.000000
50%    18.500000     2.000000            10.165000        34.500000
75%    27.250000     2.000000            14.060000        40.000000
max    36.000000     5.000000            23.300000        48.000000

In [1]:
#para dados tipo texto
import pandas as pd 

dados = pd.read_csv('TabelaSalarial.csv')

descricao_texto = dados[['Grau de Instrução', 'Região de Procedência']].describe()
print(descricao_texto)


       Grau de Instrução Região de Procedência
count                 36                    36
unique                 3                     3
top         ensino médio                 outra
freq                  18                    13


### 1. o que faz o código acima (dados.describe())?

.describe() gera um resumo estatístico descritivo para as colunas do documento.csv.

###  2. Quais informações são apresentadas?
As informações apresentadas são:

* count: O número de entradas não nulas para cada coluna.
* mean: A média dos valores em cada coluna.
* std: O desvio padrão, que mede a dispersão dos valores em torno da média.
* min: O valor mínimo encontrado na coluna.
* 25%: O primeiro quartil, que é o ponto abaixo do qual 25% dos dados caem.
* 50% (mediana): O valor central dos dados (50% dos dados estão abaixo deste valor).
* 75%: O terceiro quartil, que é o ponto abaixo do qual 75% dos dados caem.
* max: O valor máximo encontrado na coluna.

### 3. Todas as variáveis têm o mesmo número de informações?
Não, nem todas as variáveis tem as mesmas informações.
N. de filhos tem menos informaçõoes que as outras variáveis.

### 4. O que significa a variável "mean" e "std"?
Mean: média
std: desvio padrão (ou dispersão) dos valores em torno da média


### 5.  Considerando o o máximo, o mínimo, a média e o "std", o que podemos inferir sobre as variáveis nº de filhos, salário e idade?
* N. de filhos: 
    - std: Apresenta um desvio padrão baixo (em comparação com os outros) (1,268028)
    - mean: Apresenta uma média de 1.65 filhos 
    - min e max: Apresenta um mínimo de 0 filhos e uma máxima de 5 filhos.
* Salário:
    - std: Apresenta um desvio padrão médio (em comparação com os outros) (4.587458	)
    - mean: Apresenta média de 11,12(dizima) salários
    - min e max: Apresenta um mínimo de 4 salários e um máximo de 23.3
* Idade:
    - std: Apresenta um alto desvio padrão (6.737422)
    - mean: Apresenta média de 34.583333
    -min e max:  Apresenta um mínimo de 20 anos e um máximo de 48 anos


### 6. Por que não aparecem as variáveis "Grau de instrução" e "Região de Procedência"?
Não aparecem porque .describe() só fornece estatísticas para colunas numéricas.

### 7. Considerando todas as análises e informações coletadas, a média é um bom indicativo para os dados?
A média é um bom indicativo, mas deve ser considerada em conjunto com outras estatísticas, como a mediana e o desvio padrão, especialmente porque as variáveis, como salário, podem ter grandes desvios que influenciam a média

### 8. Calcule a mediana e a moda dos dados apresentados na saída dados.describe(). Os valores encontrados são próximos à média?

In [1]:
import pandas as pd
import matplotlib
dados = pd.read_csv('TabelaSalarial.csv')
dados.median()

/tmp/ipykernel_14383/3521743068.py:4: FutureWarning: The default value of numeric_only in DataFrame.median is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  dados.median()


N                      18.500
N de Filhos             2.000
Salario (x Sal Min)    10.165
Idade (em anos)        34.500
dtype: float64

In [4]:
import pandas as pd
import matplotlib
dados = pd.read_csv('TabelaSalarial.csv')
dados.mode()

N Estado Civil Grau de Instrução  N de Filhos  Salario (x Sal Min)  \
0    1       casado      ensino médio          2.0                 4.00   
1    2          NaN               NaN          NaN                 4.56   
2    3          NaN               NaN          NaN                 5.25   
3    4          NaN               NaN          NaN                 5.73   
4    5          NaN               NaN          NaN                 6.26   
5    6          NaN               NaN          NaN                 6.66   
6    7          NaN               NaN          NaN                 6.86   
7    8          NaN               NaN          NaN                 7.39   
8    9          NaN               NaN          NaN                 7.44   
9   10          NaN               NaN          NaN                 7.59   
10  11          NaN               NaN          NaN                 8.12   
11  12          NaN               NaN          NaN                 8.46   
12  13          NaN               NaN          NaN                 8.74   
13  14          NaN               NaN          NaN                 8.95   
14  15          NaN               NaN          NaN                 9.13   
15  16          NaN               NaN          NaN                 9.35   
16  17          NaN               NaN          NaN                 9.77   
17  18          NaN               NaN          NaN                 9.80   
18  19          NaN               NaN          NaN                10.53   
19  20          NaN               NaN          NaN                10.76   
20  21          NaN               NaN          NaN                11.06   
21  22          NaN               NaN          NaN                11.59   
22  23          NaN               NaN          NaN                12.00   
23  24          NaN               NaN          NaN                12.79   
24  25          NaN               NaN          NaN                13.23   
25  26          NaN               NaN          NaN                13.60   
26  27          NaN               NaN          NaN                13.85   
27  28          NaN               NaN          NaN                14.69   
28  29          NaN               NaN          NaN                14.71   
29  30          NaN               NaN          NaN                15.99   
30  31          NaN               NaN          NaN                16.22   
31  32          NaN               NaN          NaN                16.61   
32  33          NaN               NaN          NaN                17.26   
33  34          NaN               NaN          NaN                18.75   
34  35          NaN               NaN          NaN                19.40   
35  36          NaN               NaN          NaN                23.30   

    Idade (em anos) Região de Procedência  
0              26.0                 outra  
1              30.0                   NaN  
2              31.0                   NaN  
3              32.0                   NaN  
4              33.0                   NaN  
5              34.0                   NaN  
6              35.0                   NaN  
7              36.0                   NaN  
8              37.0                   NaN  
9              40.0                   NaN  
10             41.0                   NaN  
11             43.0                   NaN  
12              NaN                   NaN  
13              NaN                   NaN  
14              NaN                   NaN  
15              NaN                   NaN  
16              NaN                   NaN  
17              NaN                   NaN  
18              NaN                   NaN  
19              NaN                   NaN  
20              NaN                   NaN  
21              NaN                   NaN  
22              NaN                   NaN  
23              NaN                   NaN  
24              NaN                   NaN  
25              NaN                   NaN  
26              NaN                   NaN